quinn-proto/src/connection.rs

```diff
@@ -830,10 +830,7 @@ impl Connection {
         if self.state.is_closed() {
             return Ok(());
         }
-        let params = TransportParameters::read(
-            Side::Server,
-            &mut io::Cursor::new(self.tls.get_quic_transport_parameters().unwrap()),
-        )?;
+        let params = self.tls.transport_parameters()?.unwrap();
```

Comment:
rustls doesn't currently report an error if the transport paramters extension is missing, so this should probably be PROTOCOL_VIOLATION (if not some TLS alert) rather than unwrap; compare how the client handles this case. Pre-existing bug, but might as well fix while we're in here.

Reply:
In fact I'll just get this myself in a followup.